unison-share-api/src/Unison/Server/Types.hs

```diff
@@ -239,21 +248,44 @@ data NamedType = NamedType
 instance ToJSON NamedType where
   toEncoding = genericToEncoding defaultOptions
 
+instance FromJSON NamedType where
+  parseJSON = genericParseJSON defaultOptions
```

Comment:
I really don't like using "generic" JSON instances, they're too susceptible to change upon refactoring or even renaming fields; but I used it here because it has to match the `ToJSON` which is currently using the generic method.

Reply:
We can make them both explicit instead

Reply:
We can, it's just a bit annoying to figure out how to match it exactly to what it's currently doing (not that hard, just a bit annoying haha).

Reply:
Yeah that's fine.